e3sm_diags/plot/zonal_mean_xy_plot.py

```diff
@@ -169,3 +111,58 @@ def plot(
     fig.suptitle(parameter.main_title, x=0.5, y=0.95, fontsize=18)
 
     _save_plot(fig, parameter)
+
+
+def _save_plot(fig: plt.figure, parameter: CoreParameter):
+    """Save the plot using the figure object and parameter configs.
+
+    This function creates the output filename to save the plot. It also
+    saves each individual subplot if the reference name is an empty string ("").
```

Comment:
Moved this function to the bottom because it is a private function. Public functions should be at the top to make it easier to read the Python file and follow the order of function references (top-down approach).

Reply:
@tomvothecoder @chengzhuzhang I'm confused as to why you're making your own `_save_plot` function here. Why wouldn't we just always use [e3sm_diags/plot/utils.py](https://github.com/E3SM-Project/e3sm_diags/blob/cdat-migration-fy24/e3sm_diags/plot/utils.py)'s `_save_plot` function?

Reply:
@forsyth2 I made this design choice for zonal_mean_xy is because the the layout of the panels and plots are largely different than the 3 panel plots for lat-lon maps. i.g., For zonal_mean_xy, it contains two panels, with each a line plot. To reuse the _save_plot function in utils.py would introduce lots of conditional code blocks, which is not ideal.

Reply:
I actually did a diff of your version and the general `e3sm_diags.plot.utils._save_plot` and only found two differences involving setting the panel config and border_padding. I updated the general version with two new args for each config and removed your version.

Here is the commit with that change: e3sm_diags/plot/utils.py

Reply:
Screenshot of the diff:

![image](https://github.com/E3SM-Project/e3sm_diags/assets/25624127/e610ecdc-ed4a-487f-97fd-524209efb6bb)

Reply:
Oh, I got it, thanks for the change @tomvothecoder . @forsyth2 good point. I misinterpreted the question, it is not ideal to have a similar _save_plot function!